Backend/src/main/java/com/SYSC4806_Group13/SYSC4806_Project/Controllers/CartItemRestController.java

```diff
@@ -28,73 +29,123 @@ public CartItemRestController(CartItemRepository cartItemRepo, ListingRepository
         this.listingRepo = listingRepo;
     }
 
+    @GetMapping("/orderHistory")
+    @ResponseBody()
+    public List<CartItem> getOrderHistoryCartItems(@CurrentUser UserPrincipal userPrincipal) {
+        return cartItemRepo.findAllByUserIdAndPurchaseDateTimeIsNotNullOrderByPurchaseDateTimeDesc(userPrincipal.getId());
```

Comment:
why is this function  name so long

Reply:
This is a limitation of Java Spring repository system. These long long names are required to get the desired functionality with the DB retrieval